tests/helpers/test_trigger.py

```diff
@@ -428,3 +438,174 @@ async def test_pluggable_action(
     remove_attach_2()
     assert not hass.data[DATA_PLUGGABLE_ACTIONS]
     assert not plug_2
+
+
+@pytest.mark.parametrize(
+    "sun_service_descriptions",
+    [
+        """
+        sun:
+          fields:
+            event:
+              example: sunrise
+              selector:
+                select:
+                  options:
+                    - sunrise
+                    - sunset
+            offset:
+              selector:
+                time: null
+        """,
+        """
+        .anchor: &anchor
+          - sunrise
+          - sunset
+        sun:
+          fields:
+            event:
+              example: sunrise
+              selector:
+                select:
+                  options: *anchor
+            offset:
+              selector:
+                time: null
+        """,
+    ],
+)
+async def test_async_get_all_descriptions(
+    hass: HomeAssistant, sun_service_descriptions: str
+) -> None:
+    """Test async_get_all_descriptions."""
+    assert await async_setup_component(hass, DOMAIN_SUN, {})
+    assert await async_setup_component(hass, DOMAIN_SYSTEM_HEALTH, {})
+    await hass.async_block_till_done()
+
+    def _load_yaml(fname, secrets=None):
+        with io.StringIO(sun_service_descriptions) as file:
+            return parse_yaml(file)
+
+    with (
+        patch(
+            "homeassistant.helpers.trigger._load_triggers_files",
+            side_effect=trigger._load_triggers_files,
+        ) as proxy_load_triggers_files,
+        patch(
+            "annotatedyaml.loader.load_yaml",
+            side_effect=_load_yaml,
+        ),
+        patch.object(Integration, "has_triggers", return_value=True),
+    ):
+        descriptions = await trigger.async_get_all_descriptions(hass)
+
+    # Test we only load triggers.yaml for integrations with triggers,
+    # system_health has no triggers
+    assert proxy_load_triggers_files.mock_calls[0][1][1] == unordered(
+        [
+            await async_get_integration(hass, DOMAIN_SUN),
+        ]
+    )
+
+    # system_health does not have services and should not be in descriptions
+    assert descriptions == {
+        DOMAIN_SUN: {
+            "fields": {
+                "event": {
+                    "example": "sunrise",
+                    "selector": {"select": {"options": ["sunrise", "sunset"]}},
+                },
+                "offset": {"selector": {"time": None}},
+            }
+        }
+    }
+
+    # Verify the cache returns the same object
+    assert await trigger.async_get_all_descriptions(hass) is descriptions
```

Comment:
I think we are missing a test for checking that the cache is updated after a new integration is loaded